codexdht/private/eth/p2p/discoveryv5/protocol.nim

```diff
@@ -499,15 +501,33 @@ proc waitNodes(d: Protocol, fromNode: Node, reqId: RequestId):
   ## on that, more replies will be awaited.
   ## If one reply is lost here (timed out), others are ignored too.
   ## Same counts for out of order receival.
+  let startTime = Moment.now()
   var op = await d.waitMessage(fromNode, reqId)
   if op.isSome:
     if op.get.kind == MessageKind.nodes:
       var res = op.get.nodes.sprs
-      let total = op.get.nodes.total
+      let
+        total = op.get.nodes.total
+        firstTime = Moment.now()
+        rtt = firstTime - startTime
+      # trace "nodes RTT:", rtt, node = fromNode
+      fromNode.registerRtt(rtt)
```

Comment:
OK so I guess you're registering the min RTT across however many nodes you issued the request to? Why don't you count the RTT for the remaining replies too?

Reply:
no, here you are already in the branch of replies to a single request, sent to a single node. Check `awaitedMessages` to see why.
The response to some request (like FindNodes) is sent in multiple UDP packets. We use the first packet to measure RTT, and subsequent ones to estimate bandwidth.